common/src/hash.rs

```diff
@@ -136,12 +137,23 @@ pub fn hash_float(value: f64) -> PyHash {
     };
     x = ((x << e) & MODULUS) | x >> (BITS32 - e);
 
-    fix_sentinel(x as PyHash * value.signum() as PyHash)
+    Some(fix_sentinel(x as PyHash * value.signum() as PyHash))
 }
 
+#[inline]
 pub fn hash_complex(value: &Complex64) -> PyHash {
-    let re_hash = hash_float(value.re);
-    let im_hash = hash_float(value.im);
+    let re_hash = (match hash_float(value.re) {
+        Some(value) => Some(value),
+        None => Some(hash_pointer(value as *const _ as *const std::ffi::c_void)),
```

Comment:
I thought the pointer to pass `hash_pointer` is `&PyObject`, not `&Complex64`

Reply:
Thanks for pointing it out.
I think the easiest way is to pass `Py<PyComplex>`.
But the function also cannot use python type in the signature.
Should I move `hash_complex` implementation from `common/src` to `complex.rs`?
Or passing address itself as a second argument is acceptable like below?

```rust
pub fn hash_complex(value: &Complex64, ptr: *const std::ffi::c_void) -> PyHash
```